lib/actor.js

```diff
@@ -109,14 +110,30 @@ function recordStep(step, args) {
 
   const task = `${step.name}: ${step.humanizeArgs()}`;
   let val;
-
+  let timeout;
```

Comment:
as far as I understand, the code could be simpler without timer id tracing and clearTimeout calls as rejection of timeout promises after completion of main action doesn't affect anything.